pkg/gui/controllers/branches_controller.go

```diff
@@ -157,36 +181,34 @@ func (self *BranchesController) setUpstream(selectedBranch *models.Branch) error
 						return self.c.Error(err)
 					}
 					return nil
-				},
-				Key: 'u',
+				})
 			},
-			{
-				LabelColumns: []string{self.c.Tr.SetUpstream},
-				OnPress: func() error {
-					return self.c.Helpers().Upstream.PromptForUpstreamWithoutInitialContent(selectedBranch, func(upstream string) error {
-						upstreamRemote, upstreamBranch, err := self.c.Helpers().Upstream.ParseUpstream(upstream)
-						if err != nil {
-							return self.c.Error(err)
-						}
+			Key: 's',
+		},
+	}
 
-						if err := self.c.Git().Branch.SetUpstream(upstreamRemote, upstreamBranch, selectedBranch.Name); err != nil {
-							return self.c.Error(err)
-						}
-						if err := self.c.Refresh(types.RefreshOptions{
-							Mode: types.SYNC,
-							Scope: []types.RefreshableView{
-								types.BRANCHES,
-								types.COMMITS,
-							},
-						}); err != nil {
-							return self.c.Error(err)
-						}
-						return nil
-					})
-				},
-				Key: 's',
+	if selectedBranch.IsTrackingRemote() {
+		options = append(options, &types.MenuItem{
+			LabelColumns: []string{self.c.Tr.ViewUpstreamResetOptions},
+			OpensMenu:    true,
+			OnPress: func() error {
+				if selectedBranch.RemoteBranchNotStoredLocally() {
+					return self.c.ErrorMsg(self.c.Tr.UpstreamNotStoredLocallyError)
+				}
+
+				err := self.c.Helpers().Refs.CreateGitResetMenu(fmt.Sprintf("%s/%s", selectedBranch.UpstreamRemote, selectedBranch.Name))
+				if err != nil {
+					return self.c.Error(err)
+				}
+				return nil
 			},
-		},
+			Key: 'g',
+		})
+	}
+
+	return self.c.Menu(types.CreateMenuOptions{
+		Title: self.c.Tr.BranchUpstreamOptionsTitle,
+		Items: options,
```

Comment:
Can we add a tooltip saying 'View options for resetting the checked-out branch onto {{upstream}}. Note: this will not reset the selected branch onto the upstream, it will reset the checked-out branch onto the upstream'

Reply:
Unfortunately, the `types.CreateMenuOptions` type does not support tooltips. Do you want me to try adding it as part of this branch?

Reply:
Yes please!

Reply:
Never mind - no changes required 😄